### PR TITLE
[Workspace] add tests for data source association and dissociation

### PR DESCRIPTION
**Pull request by raintygao**
### Description

add tests for data source association and dissociation

### Screenshot
image

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Hailong-am — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+const miscUtils = new MiscUtils(cy);
+const workspaceName = 'test_workspace_collaborators';
+let workspaceId;
+let dataSourceTitle1 = 'no_auth_data_source_title_1';
+let dataSourceTitle2 = 'no_auth_data_source_title_2';
+let dataSourceId1;
+let dataSourceId2;
+if (
+  Cypress.env('WORKSPACE_ENABLED') &&
+  Cypress.env('SECURITY_ENABLED') &&
```
Comment (SuZhou-Joe) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js:16]: Do we need security enabled for testing data source association?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    after(() => {
+      cy.deleteDataSource(dataSourceId1);
+      cy.deleteDataSource(dataSourceId2);
+    });
+    afterEach(() => {
+      cy.deleteWorkspaceById(workspaceId);
+    });
+
+    it('should associate and dissociate data source successfully', () => {
```
Comment (SuZhou-Joe) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js:53]: Do we need to add test cases for associating DQC?

Comment (raintygao) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js:53]: We don't have related infra of this type data source for now, may need to figure out a way to create this type data source first.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js (hunk 1 above)
Comment (raintygao) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js:16]: Do you mean we don't need this verification?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js (hunk 2 above)
Comment (raintygao) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js:53]: We don't have related infra of this type data source for now, may need to figure out a way to create this type data source with related team first.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**